halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/DeploymentEnvironmentService.java

```diff
@@ -56,14 +62,26 @@ public DeploymentEnvironment getDeploymentEnvironment(String deploymentName) {
 
   public void setDeploymentEnvironment(String deploymentName, DeploymentEnvironment newDeploymentEnvironment) {
     DeploymentConfiguration deploymentConfiguration = deploymentService.getDeploymentConfiguration(deploymentName);
+
+    String defaultLocation = lookupDefaultLocation(deploymentName, newDeploymentEnvironment.getAccountName());
+    String location = newDeploymentEnvironment.getLocation();
+    newDeploymentEnvironment.setLocation(StringUtils.isEmpty(location) ? defaultLocation : location);
```

Comment:
It's OK to change the location with the account - but if the user hasn't overriden the location yet they shouldn't have to set it each time after the first edit to the account name.